src/server/schema/collection.schema.ts

```diff
@@ -0,0 +1,30 @@
+import { z } from 'zod';
+import { isDefined } from '~/utils/type-guards';
+import { CollectionContributorPermission } from '@prisma/client';
+
+export type AddCollectionItemInput = z.infer<typeof addCollectionItemInputSchema>;
+
+export const addCollectionItemInputSchema = z
+  .object({
+    articleId: z.number().optional(),
+    postId: z.number().optional(),
+    modelId: z.number().optional(),
+    imageId: z.number().optional(),
+    collectionIds: z.number().array(),
```

Comment:
Let's add a constraint here to always require at least one item in the id list
```suggestion
    collectionIds: z.number().array().min(1, 'Please select at least one collection'),
```